src/lib/php/Report/ClearedGetterCommon.php

```diff
@@ -163,26 +170,39 @@ protected function groupStatements($ungrupedStatements, $extended, $agentcall, $
           $statements[] = $singleStatement;
         }
       }
-      if (!empty($statement['textfinding']) && !empty($agentcall)) {
-        $findings[$fileName] = array(
+      if (!empty($statement['textfinding']) && !empty($agentCall) && $agentCall != "license") {
+        $findings[] = array(
+            "licenseId" => $licenseId,
             "content" => convertToUTF8($statement['textfinding'], false),
             "text" => convertToUTF8($text, false),
             "files" => array($fileName)
           );
         if ($extended) {
-          $findings[$fileName]["comments"] = convertToUTF8($comments, false);
+          $key = array_search($statement['textfinding'], array_column($findings, 'content'));
+          $findings[$key]["comments"] = convertToUTF8($comments, false);
         }
       }
+      //To keep the schedular alive for large files
+      $countLoop += 1;
+      if ($countLoop % 500 == 0) {
+        $thousandLoop++;
+        $objectAgent->heartbeat(1);
```

Comment:
I was just wondering if we can send `0` instead of `1`. I am not sure if this will be accepted by scheduler but it will help you keep the actual count to number of files processed.

Any inputs on that @shaheemazmalmmd ?